.changeset/few-emus-walk.md

```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": patch
+---
+
+`SideNav::List::Title`, `SideNav::List::BackLink`: fixed issue with long text strings without spaces not wrapping
```

Comment:
Also the "List::Link" is impacted (it uses `hds-side-nav__list-item-text` inside)

```suggestion
`SideNav::List::Title`, `SideNav::List::BackLink`, `SideNav::List::Link`: fixed issue with long text strings without spaces not wrapping
```